src/Storages/Kafka/ReadBufferFromKafkaConsumer.cpp

```diff
@@ -302,102 +303,131 @@ void ReadBufferFromKafkaConsumer::resetToLastCommitted(const char * msg)
-            /// Don't drop old messages immediately, since we may need them for virtual columns.
-            auto new_messages = consumer->poll_batch(batch_size, std::chrono::milliseconds(poll_timeout));
-
-            if (stopped)
+            return false;
+        }
+        else if (stalled_status == REBALANCE_HAPPENED)
+        {
+            if (!new_messages.empty())
             {
-                was_stopped = true;
-                offsets_stored = 0;
-                return false;
-            }
-            else if (rebalance_happened)
-            {
-                if (!new_messages.empty())
-                {
-                    // we have polled something just after rebalance.
-                    // we will not use current batch, so we need to return to last commited position
-                    // otherwise we will continue polling from that position
-                    resetToLastCommitted("Rewind last poll after rebalance.");
-                }
-
-                offsets_stored = 0;
-                return false;
+                // we have polled something just after rebalance.
+                // we will not use current batch, so we need to return to last commited position
+                // otherwise we will continue polling from that position
+                resetToLastCommitted("Rewind last poll after rebalance.");
             }
+            return false;
+        }
 
-            if (new_messages.empty())
+        if (new_messages.empty())
+        {
+            // While we wait for an assignment after subscription, we'll poll zero messages anyway.
+            // If we're doing a manual select then it's better to get something after a wait, then immediate nothing.
+            if (assignment.empty())
             {
-                // While we wait for an assignment after subscription, we'll poll zero messages anyway.
-                // If we're doing a manual select then it's better to get something after a wait, then immediate nothing.
-                if (assignment.empty())
+                waited_for_assignment += poll_timeout; // slightly innaccurate, but rough calculation is ok.
+                if (waited_for_assignment < MAX_TIME_TO_WAIT_FOR_ASSIGNMENT_MS)
                 {
-                    waited_for_assignment += poll_timeout; // slightly innaccurate, but rough calculation is ok.
-                    if (waited_for_assignment < MAX_TIME_TO_WAIT_FOR_ASSIGNMENT_MS)
-                    {
-                        continue;
-                    }
-                    else
-                    {
-                        LOG_TRACE(log, "Can't get assignment");
-                        stalled = true;
-                        return false;
-                    }
-
+                    continue;
                 }
                 else
                 {
-                    LOG_TRACE(log, "Stalled");
-                    stalled = true;
+                    LOG_TRACE(log, "Can't get assignment");
```

Comment:
While you are are here, can you change the level of this message from trace to at least warning (or even better error)?

Reply:
That makes sence. Will do.